From 00cce3fb80a8b6514648f01cda8c382cee784fb3 Mon Sep 17 00:00:00 2001
From: Shirish S <shirish.s@amd.com>
Date: Thu, 16 Mar 2017 15:28:41 +0530
Subject: [PATCH 0898/2094] drm: add check for plane functions

update_plane() and disable_plane() functions
assoiciated with setting plane are called
without any check, causing kernel panic.

This patch adds the required check to avoid it.

Signed-off-by: Shirish S <shirish.s@amd.com>
---
 drivers/gpu/drm/drm_plane.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/drm_plane.c b/drivers/gpu/drm/drm_plane.c
index 249c0ae..6b6fc77 100644
--- a/drivers/gpu/drm/drm_plane.c
+++ b/drivers/gpu/drm/drm_plane.c
@@ -456,6 +456,12 @@ static int __setplane_internal(struct drm_plane *plane,
 {
 	int ret = 0;
 
+	if (plane->funcs->disable_plane == NULL ||
+				 plane->funcs->update_plane == NULL) {
+		DRM_ERROR("plane funcs not implemented\n");
+		ret = -EPERM;
+		goto out;
+	}
 	/* No fb means shut it down */
 	if (!fb) {
 		plane->old_fb = plane->fb;
-- 
2.7.4

